tests/test_packsmanager.py

```diff
@@ -92,3 +99,218 @@ def test_tmp_file_structure(input, expected, example_cases):
+    # 4) copy one example (unambiguous)
+    # 5) copy list of examples from same pack (unambiguous)
+    # 6) copy list of examples from different pack (unambiguous)
+    # 7) copy all examples from a pack
+    # 8) copy all examples from list of packs
+    # 9) copy all examples from all packs
+    (  # 1) copy one example, (ambiguous)
+        ["ex1"],
+        [
+            Path("packA/ex1/path1/script1.py"),
+            Path("packB/ex1/path2/script2.py"),
+        ],
+    ),
+    (  # 2) copy list of examples from same pack (ambiguous)
+        ["ex1", "ex2"],
+        [
+            Path("packA/ex1/path1/script1.py"),
+            Path("packB/ex1/path2/script2.py"),
+            Path("packA/ex2/script3.py"),
+        ],
+    ),
+    (  # 3) copy list of examples from different packs (ambiguous)
+        ["ex1", "ex1"],
+        [
+            Path("packA/ex1/path1/script1.py"),
+            Path("packB/ex1/path2/script2.py"),
+        ],
+    ),
+    (  # 4) copy one example (unambiguous)
+        ["ex2"],
+        [
+            Path("packA/ex2/script3.py"),
+        ],
+    ),
+    (  # 5) copy list of examples from same pack (unambiguous)
+        ["ex3", "ex4"],
+        [
+            Path("packB/ex3/script4.py"),
+            Path("packB/ex4/script5.py"),
+        ],
+    ),
+    (  # 6) copy list of examples from different packs (unambiguous)
+        ["ex2", "ex3"],
+        [
+            Path("packA/ex2/script3.py"),
+            Path("packB/ex3/script4.py"),
+        ],
+    ),
+    (  # 7) copy all examples from a pack
+        ["packA"],
+        [
+            Path("packA/ex1/path1/script1.py"),
+            Path("packA/ex2/script3.py"),
+        ],
+    ),
+    (  # 8) copy all examples from list of packs
+        ["packA", "packB"],
+        [
+            Path("packA/ex1/path1/script1.py"),
+            Path("packA/ex2/script3.py"),
+            Path("packB/ex1/path2/script2.py"),
+            Path("packB/ex3/script4.py"),
+            Path("packB/ex4/script5.py"),
+        ],
+    ),
+    (  # 9) copy all examples from all packs
+        ["all"],
+        [
+            Path("packA/ex1/path1/script1.py"),
+            Path("packA/ex2/script3.py"),
+            Path("packB/ex1/path2/script2.py"),
+            Path("packB/ex3/script4.py"),
+            Path("packB/ex4/script5.py"),
+        ],
+    ),
+]
+
+
+# input: list of str - cli input(s) to copy_examples
+# expected_paths: list of Path - expected relative paths to copied examples
+@pytest.mark.parametrize("input,expected_paths", copy_params)
+def test_copy_examples(input, expected_paths, example_cases):
+    examples_dir = example_cases / "case5"
+    pm = PacksManager(root_path=examples_dir)
+    target_dir = example_cases / "user_target"
+    pm.copy_examples(input, target_dir=target_dir)
+    actual = list(target_dir.rglob("*"))
+    expected = [target_dir / path for path in expected_paths]
+    assert actual == expected
```

Comment:
to be on the safe side I would sort these in the assert, so `assert actual.sort() ...`  or `sorted(actual)` (I don't remember which one does it in place)